### PR TITLE
Week 16, Project Labyrinth - Anna Lenksjö & Isabella Markskog

### PR DESCRIPTION
**Pull request by IsabellaM5**
**Link to our deployed project**
https://goofy-dijkstra-c1398c.netlify.app/ 

**Our reflections on how this week's project turned out**
We did all the blue level requirements and some stretch goals.

The first two days were a bit hard to get into and grasp the concept of using thunks in API calls. Also structuring the POST request body to get the desired API response was tricky. After figuring out how to do this, we felt the rest of the days went relatively smooth. We found a picture that we really liked which we adapted the rest of the styling to and we're so happy with the end result! 🤩 All in all, a fun project!

**Things we'd like to have clarified or explained in more detail**
We would've liked a more detailed description of the concept of this labyrinth game to get a more feel for it. It took some time to get into the thinking of how to structure it.

Other than that, perhaps more explanation about localStorage and the functionality for it.

/Anna & Isabella 🐯🐯

### CODE REVIEW

**Review by ElleFrankenberg — APPROVED** [review]
Great work!!! You definitely fullfilled the blue requirements and more💥 Clean code and really good naming.  Nice design on both tablet and desktop! 
The only comment that I have is when I viewed this on my iPhone the input field was a bit to long to fit the screen.  Besides this it is all 🙌

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
After I give my character name I have instantly seen this page: image
Refreshing the page doesn't change anything. If you are using local storage to persist your data please remember to give your user possibility to clean it (Restart button for example)

**Comment by IsabellaM5** [comment]
Hi @KarolCislowski 

Please see our above message. 

/Anna & Isabella

File: code/src/App.js [hunk 1]
```diff
@@ -1,9 +1,32 @@
...
+
+const GlobalStyle = createGlobalStyle`
+  body {
+    background-image: url('/assets/desktop-image.jpg');
+    background-size: cover;
+    background-repeat: no-repeat;
+    background-position: center;
+    font-family: 'New Tegomin', serif;
+  }
+`
```
Comment (KarolCislowski) [inline code/src/App.js:17]: nice global tyle 👍🏻

File: code/src/components/Content.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+`
+
+const Heading = styled.h1`
+  color: rgb(246,226,189);
+`
+
+const CustomButton = styled.button`     
+  border-radius: 6px;
+  border: 2px solid #3b2e36;
+  background-color: rgb(82,88,13, 0.7);
```
Comment (KarolCislowski) [inline code/src/components/Content.js:23]: you should specify `rgba` when you are using the alpha canal.

File: code/src/components/Content.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  return (
+    <Main>
+      {error 
+        ? (<>
+            <Heading>Something went wrong, reason: {error}</Heading>
+            <CustomButton onClick={onRestartButton} >
+              RESTART
+            </CustomButton>
```
Comment (KarolCislowski) [inline code/src/components/Content.js:58]: Great idea 👍🏻

File: code/src/components/GameCard.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  return (
+    <>
+      <Container>
+          <DescriptionText>{action.description}</DescriptionText>
+          <CustomButton onClick={() => onSetDirection(action)}>HEAD {action.direction.toUpperCase()}</CustomButton>
+      </Container>
+    </>
+  )
```
Comment (KarolCislowski) [inline code/src/components/GameCard.js:79]: you do not need <></> here 🤓

File: code/src/components/GameScreen.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
+import React from 'react'
+import styled from "styled-components/macro"
+import { useSelector, useDispatch } from 'react-redux'
+
+import labyrinth from 'reducers/labyrinth'
+import { GameCard } from './GameCard' 
+
+const Container = styled.div`
+  box-shadow: 0px 2px 1px -1px rgb(0 0 0 / 20%),
+    0px 1px 1px 0px rgb(0 0 0 / 14%), 0px 1px 3px 0px rgb(0 0 0 / 12%);
```
Comment (KarolCislowski) [inline code/src/components/GameScreen.js:10]: Nice 👍🏻

File: code/src/components/StartScreen.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+            autoFocus
+          />
+          <CustomButton
+            onClick={onUserSet}
+          >
+            START
+          </CustomButton>
+        </InputContainer>
+      </Container>
+    </>
```
Comment (KarolCislowski) [inline code/src/components/StartScreen.js:136]: another misplaced use of the <></> 🤓

File: code/src/reducers/labyrinth.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+  name: 'labyrinth',
+  initialState: {
+    playerName: '',
+    actions: localStorage.getItem('startContent')
+      ? JSON.parse(localStorage.getItem('startContent'))
+      : [],
+    description: '',
+    history: [],
+    error: '',
+    loading: false
```
Comment (KarolCislowski) [inline code/src/reducers/labyrinth.js:13]: Nice initial state 👍🏻

File: code/src/reducers/labyrinth.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        }
+      })
+      .then(data => {
+        dispatch(labyrinth.actions.setActions(data))
+        localStorage.setItem('startContent', JSON.stringify(data))
+      })
+      .catch(error => dispatch(labyrinth.actions.setError(error.message)))
+      .finally(() => dispatch(labyrinth.actions.setLoading(false)))
+    }
+}
```
Comment (KarolCislowski) [inline code/src/reducers/labyrinth.js:99]: 💯

**Review by KarolCislowski — APPROVED** [review]
Really nice game 🎮

**Comment by KarolCislowski** [comment]
Hello, it's your project. Do you remember me? I may be imperfect, but I believe you can still change that before you sit down to your final masterpiece!
image

**Pull request closed without merging**